wgpu/src/lib.rs

```diff
@@ -379,6 +380,13 @@ impl Drop for Sampler {
 pub type SurfaceConfiguration = wgt::SurfaceConfiguration<Vec<TextureFormat>>;
 static_assertions::assert_impl_all!(SurfaceConfiguration: Send, Sync);
 
+trait WgpuSurfaceRequirement: HasWindowHandle + HasDisplayHandle + WasmNotSend + WasmNotSync {}
```

Comment:
We need to make this public and document this. Otherwise an end user would have no way to know how to get a T which implements this.